jerry-core/jerry.c

```diff
@@ -2166,10 +2214,27 @@ jerry_value_to_string (const jerry_value_t value) /**< input value */
 {
   jerry_assert_api_available ();
 
+  if (ECMA_IS_VALUE_ERROR (value))
+  {
+    return jerry_create_object_value (jerry_create_error (JERRY_ERROR_TYPE, error_value_msg_p));
+  }
+
   return ecma_op_to_string (value);
 } /* jerry_value_to_string */
 
 /**
+ * Convert Jerry value to normal value
```

Comment:
"Remove the error flag" sounds better.

I think we should emphasize that error is a flag, which can be combined with any value.